### PR TITLE
Add option --todo-only

### PR DESCRIPTION
**Pull request by perlpunk**
In this view we only want to see TODO items in a compact list

Issue: https://progress.opensuse.org/issues/94762

### CODE REVIEW

File: openqa_review/openqa_review.py [hunk 1]
```diff
@@ -1023,6 +1029,10 @@ def _get_bugref_for_softfailed_module(self, result_item, module_name):
                 % absolute_url(self.root_url, result_item)
             )
 
+    def has_todo_issues(self):
+        """Return tru if Report has new or existing todo issues."""
+        return True if (self.issues["new"]["todo"] or self.issues["existing"]["todo"]) else False
```
Comment (okurz) [inline openqa_review/openqa_review.py:1034]:
```suggestion
        """Tell if report has new or existing todo issues."""
        return self.issues["new"]["todo"] or self.issues["existing"]["todo"]
```

File: openqa_review/openqa_review.py [hunk 2]
```diff
@@ -1033,19 +1043,31 @@ def _todo_issues_str(self):
...
-        return todo_issues if (self.issues["new"]["todo"] or self.issues["existing"]["todo"]) else ""
 
     def __str__(self):
         """Return as markdown."""
         abbrev = self.args.abbreviate_test_issues
+        if self.args.todo_only:
+            if not self.has_todo_issues():
+                return ""
+
+            return self._todo_issues_str()
```
Comment (okurz) [inline openqa_review/openqa_review.py:1069]: how about

```suggestion
            return self._todo_issues_str() if self.has_todo_issues() else ""
```

or at least

```suggestion
            if self.has_todo_issues():
                self._todo_issues_str()    
            return ""
```

File: openqa_review/openqa_review.py [hunk 3]
```diff
@@ -1299,6 +1326,13 @@ def parse_args():
         dest="skip_passed",
         help="Skip passed reports",
     )
+    parser.add_argument(
+        "--todo-only",
+        action="store_true",
+        default=False,
+        dest="todo_only",
+        help="List inly TODO items as one liners",
```
Comment (okurz) [inline openqa_review/openqa_review.py:1334]: s/inly/only/

File: openqa_review/openqa_review.py [hunk 4]
```diff
@@ -1423,9 +1457,16 @@ def _next_label(self):
...
+            report_str = "**TODO: review**\n\n"
         for k, v in self.report.items():
             if not self.args.skip_passed or type(v) is not ProductReport or not v.is_passed():
-                report_str += "# %s\n\n%s\n---\n" % (k, v)
+                generated_report = str(v)
+                if len(generated_report):
+                    if self.args.todo_only:
+                        report_str += "* %s:\n%s" % (k, generated_report)
+                    else:
+                        report_str += "# %s\n\n%s\n---\n" % (k, generated_report)
```
Comment (okurz) [inline openqa_review/openqa_review.py:1469]: hm, this becomes a bit convoluted. Let's see. How about:

```suggestion
                if not len(generated_report):
                    continue
                report_str += "* %s:\n%s" % (k, generated_report) if self.args.todo_only else "# %s\n\n%s\n---\n" % (k, generated_report)
```

**Review by okurz — CHANGES_REQUESTED** [review]
You are adding a lot of big test data. Please strip it down to the data that is needed to pass the tests

File: openqa_review/openqa_review.py (hunk 1 above)
Comment (perlpunk) [reply openqa_review/openqa_review.py]: done

File: openqa_review/openqa_review.py (hunk 2 above)
Comment (perlpunk) [reply openqa_review/openqa_review.py]: done

File: openqa_review/openqa_review.py (hunk 3 above)
Comment (perlpunk) [reply openqa_review/openqa_review.py]: done

File: openqa_review/openqa_review.py (hunk 4 above)
Comment (perlpunk) [reply openqa_review/openqa_review.py]: done

**Comment by perlpunk** [comment]
> You are adding a lot of big test data. Please strip it down to the data that is needed to pass the tests

Ok, I removed some parts in the JSON files

**Comment by perlpunk** [comment]
I don't know how to fix the last issue.
https://app.travis-ci.com/github/os-autoinst/openqa_review/jobs/524734093
```
./openqa_review/openqa_review.py:1050:34: BLK100 Black would make changes.
```
It's nice to know that black would make changes, however it would be nicer to know which changes.

I can't find information on how to run this locally.

Calling `tox` results in a lot of errors.

`pip3 install -r requirements.txt` also results in an error
```
  Running setup.py develop for openqa-review
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/tina/openqa-devel/repos/openqa_review/setup.py'"'"'; __file__='"'"'/home/tina/openqa-devel/repos/openqa_review/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix=
         cwd: /home/tina/openqa-devel/repos/openqa_review/
    Complete output (30 lines):
    running develop
    /tmp/pip-build-env-uu0w1k6a/overlay/lib/python3.6/site-packages/setuptools/dist.py:502: UserWarning: The version specified ('1.25.0-15-g774e790') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
      "details." % version
    WARNING: The user site-packages directory is disabled.
    error: can't create or remove files in install directory
    
    The following error occurred while trying to add or remove files in the
    installation directory:
    
        [Errno 13] Permission denied: '/usr/lib/python3.6/site-packages/test-easy-install-4424.write-test'
    
    The installation directory you specified (via --install-dir, --prefix, or
    the distutils default setting) was:
    
        /usr/lib/python3.6/site-packages/
    
    Perhaps your account does not have write access to this directory?  If the
    installation directory is a system-owned directory, you may need to sign in
    as the administrator or "root" account.  If you do not have administrative
    access to this machine, you may wish to choose a different installation
    directory, preferably one that is listed in your PYTHONPATH environment
    variable.
    
    For information on other options, you may wish to consult the
    documentation at:
    
      https://setuptools.readthedocs.io/en/latest/easy_install.html
    
    Please make the appropriate changes for your system and try again.
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/tina/openqa-devel/repos/openqa_review/setup.py'"'"'; __file__='"'"'/home/tina/openqa-devel/repos/openqa_review/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix= Check the logs for full command output.
```

**Comment by perlpunk** [comment]
Calling `flake8 openqa_review/openqa_review.py` gives no output at all and a zero exit code.

**Comment by perlpunk** [comment]
Aha, `.tox/black/bin/black openqa_review/openqa_review.py` reformatted the file for me

File: openqa_review/openqa_review.py [hunk 5]
```diff
@@ -1023,6 +1029,10 @@ def _get_bugref_for_softfailed_module(self, result_item, module_name):
                 % absolute_url(self.root_url, result_item)
             )
 
+    def has_todo_issues(self):
+        """Tell if report has new or existing todo issues."""
+        return True if (self.issues["new"]["todo"] or self.issues["existing"]["todo"]) else False
```
Comment (okurz) [inline openqa_review/openqa_review.py:1034]: I also suggested to change this line. Just return the result of the evaluation directly, don't return True/False

File: tests/test_openqa_review.py [hunk 6]
```diff
@@ -418,6 +419,17 @@ def test_skip_passed():
...
 
+def test_todo_only():
+    args = cache_test_args_factory()
+    args.todo_only = True
+    args.show_empty = False
+    args.load_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "todo-only")
+    args.job_group_urls = ",".join(map(lambda i: "https://openqa.opensuse.org/group_overview/%i" % i, (2, 24, 35)))
+    args.arch = None
+    report = str(openqa_review.generate_report(args))
+    compare_report(report, os.path.join(args.load_dir, "report-2-24-35.md"))
```
Comment (okurz) [inline tests/test_openqa_review.py:430]: I suggest to rename the report `report-todo_only.md` or similar

**Review by okurz — CHANGES_REQUESTED** [review]
Hm, some JSON files seem to be really big still. Also please remove more from the html documents. As you now have a working test that should be trivial to do. Remove lines, run the test, continue

File: tests/test_openqa_review.py (hunk 6 above)
Comment (perlpunk) [reply tests/test_openqa_review.py:430]: done

File: openqa_review/openqa_review.py (hunk 5 above)
Comment (perlpunk) [reply openqa_review/openqa_review.py]: done

**Comment by perlpunk** [comment]
> Hm, some JSON files seem to be really big still. Also please remove more from the html documents.

I think I removed pretty much everything possible now

**Comment by perlpunk** [comment]
https://app.travis-ci.com/github/os-autoinst/openqa_review/jobs/524978936#L223
```
File "/home/travis/build/os-autoinst/openqa_review/.tox/flake8/lib/python3.6/site-packages/flake8_black.py", line 11, in <module>
224    import toml
225ModuleNotFoundError: No module named 'toml'
```
What am I supposed to do here?

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
I don't know about the toml problem. Maybe recent dependency problem of black?

**Comment by perlpunk** [comment]
I force-pushed, and now it passed...

**Pull request merged**